src/main/java/edu/hawaii/its/api/service/GroupingAttributeService.java

```diff
@@ -59,6 +62,14 @@ public GroupingAttributeService(GrouperService grouperService,
         this.timestampService = timestampService;
     }
 
+    public boolean isGroupingPath(String path) {
+        try {
+            return isGroupAttribute(path, TRIO);
+        } catch (Exception e) {
+            return false;
+        }
+    }
```

Comment:
Just curious, was `groupingsService.isGroupingPath()` not sufficient enough?

Reply:
From what I understood, Eddy and Julio are not sure if the function from `GroupPathService.isGroupingPath()` checks the TRIO attribute. 

The reason why I am using try-catch is because this function returns a server error if the path specified doesn't refer to a grouping nor a group. I can get rid of it, if it is a bad practice.

Reply:
So would it make more sense to call the endpoint isTRIO?

Reply:
No, you can't have a grouping which is not TRIO.

Reply:
Could you make the following changes? Thanks!:
1. Remove the try-catch:
```
public boolean isGroupingPath(String path) {
    return isGroupAttribute(path, TRIO);
}
```
2. Update `isGroupAttribute` to use null guard:
```
public boolean isGroupAttribute(String groupPath, String attributeName) {
    GroupAttributeResults groupAttributesResults = grouperService.groupAttributeResults(attributeName, groupPath);

    if (groupAttributesResults == null) {
        return false;
    }

    return groupAttributesResults.getGroupAttributes().stream()
            .anyMatch(groupAttribute -> groupAttribute.getAttributeName().equals(attributeName));
}
```
3. Move the code to `GroupPathService.isGroupingPath(String path)`, replacing it entirely since it is not called anywhere
4. Remove `isGroupingPathTest` from `TestGroupingAttributeService` because `TestGroupPathService` already contains a duplicate test.

Reply:
Good suggestions, done.

Reply:
The problem with this implementation is that I am getting a beans' dependencies cycle error: 

- GroupPathService needs GroupingAttributeService bean
- GroupingAttributeService needs GroupingsService bean
- GroupingsService needs GroupPathService bean

Reply:
Ok then move the method back to GroupingAttributeService but add a comment above saying that it checks TRIO attribute.

Reply:
Is using @Lazy or Bean injection through a setter a bad practice?

Reply:
Yeah I would say so. When we still used `@Autowired` above the services, this dependency cycle error would not show up.